packages/utils/src/url.ts

```diff
@@ -18,3 +18,10 @@ export function isValidHttpUrl(urlStr: string): boolean {
 
   return url.protocol === "http:" || url.protocol === "https:";
 }
+
+/**
+ * Sanitize URL to prevent leaking user credentials in logs
+ */
+export function toSafePrintableUrl(urlStr: string): string {
+  return new URL(urlStr).origin;
```

Comment:
This will strip out `search`part of URL, assuming this is fine.

Reply:
This is desired in case someone provides secrets in query params. I don't know of anybody that does this but we can't prevent users from doing that and passing credentials directly as part of the URL is common because it works with all clients.

Reply:
Got it, somehow I assumed credentials could be passed via URL username/password properties

Reply:
In case of rescue node, they are passing the user credentials in the [userinfo](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.1) part of the URL.

Reply:
Good catch! Completely forgot about that, I added a note to the function now and moved all URL logs after they are already validated by http / jsonrpc  client.

Also looked into sanitizing invalid URLs (in catch block) but that got too messy.